projects/plugins/jetpack/class.jetpack.php

```diff
@@ -2435,19 +2447,36 @@ public static function get_available_modules( $min_version = false, $max_version
 				continue;
 			}
 
+			$mod_details = self::get_module( $slug );
+
+			if ( is_bool( $requires_connection ) && $requires_connection !== $mod_details['requires_connection'] ) {
+				continue;
+			}
+
+			if ( is_bool( $requires_user_connection ) && $requires_user_connection !== $mod_details['requires_user_connection'] ) {
+				continue;
+			}
```

Comment:
```suggestion
			if ( null !== $requires_connection && (bool) $requires_connection !==
$mod_details['requires_connection'] ) {
				continue;
			}

			if ( null !== $requires_user_connection && (bool) $requires_user_connection !==
$mod_details['requires_user_connection'] ) {
				continue;
			}
```
I believe we should either throw an exception (or return a `WP_Error` object) if we enforce strict typing inside the function, or implicitly convert the values to boolean.
Passing `1` or `"true"` instead of a boolean `true` is definitely a bad coding practice, and I myself am a huge proponent of strict typing, but we can't silently filter out a "soft-typing" value that's supposed to be converted to a boolean, and silently ignore it.
So I would suggest either making strict typing explicit, or let PHP convert the types.